client/rpc/src/state/mod.rs

```diff
@@ -360,10 +379,98 @@ impl<Block, Client> State<Block, Client>
 			}.boxed()
 		})?;
 
+		module.register_subscription(
+			"state_runtimeVersion",
+			"state_unsubscribeRuntimeVersion",
+			|_params, mut sink, ctx| {
+				let executor = ctx.executor.clone();
+				let client = ctx.client.clone();
+
+				let mut previous_version = client.runtime_version_at(&BlockId::hash(client.info().best_hash))
+					.expect("best hash is valid; qed");
+				let _ = sink.send(&previous_version);
+				let rt_version_stream = client.storage_changes_notification_stream(Some(&[StorageKey(well_known_keys::CODE.to_vec())]), None, )
+					.map_err(|blockchain_err| Error::Client(Box::new(blockchain_err)))
+					.map_err(to_jsonrpsee_call_error)?;
+
+				let fut = async move {
+					let mut stream = rt_version_stream
+						.filter_map(move |_| {
+							let info = client.info();
+							let version = client
+								.runtime_version_at(&BlockId::hash(info.best_hash));
+								match version {
+									Ok(v) => if previous_version != v {
+											previous_version = v.clone();
+											future::ready(Some(v))
+										} else {
+											future::ready(None)
+										},
+									Err(e) => {
+										log::error!("Could not fetch current runtime version. Error={:?}", e);
+										// TODO: Does this terminate the stream? What is the best way to let users know?
```

Comment:
yeah that is correct.

you could either just drop sink then the server will send `Subscription closed` the client or you just do `sink.close("your message")`;

Reply:
I can't `drop` or call `sink.close()` here unfortunately, as the `sink` is borrowed in the `stream.for_each` block below. :/

Reply:
right, maybe we could implement `Clone` for `SubscriptionSink` then it should work?

I was a little afraid of making `fn send(&mut self)` let's see if it will be an issue should quite easy to change though but a breaking change ^^

Reply:
After think about this, it's not correct to terminate the sender here (but it's doesn't work either) because we are only doing filter_map and we want to actually send the potential previous elements on the stream before terminating it.

Thus, the error should sent in  `take_while().for_each()` below but I doesn't work either, I think we need to either implement `Clone` for the Sink and/or make the methods on it `&self` instead `&mut self`.

k?